docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md

```diff
@@ -39,16 +39,57 @@ App Service Web Apps offers a fully managed platform optimized for hosting web a
+
+![Migrate containerized on-premises .NET application to Azure Web Apps for Containers](./media/image1-8.png)
+
+This approach also works well if your development team is able to move to a container-based development process. The "inner loop" of developing apps with containers includes building the app with containers. Changes made to the code as well as to container configuration are pushed to source control, and an automated build is responsible for publishing new container images to a registry like Docker Hub or Azure Container Registry. These images are then used as the basis for additional development, as well as for deployments to production, as shown in the following diagram:
+
+![End to End Docker DevOps Lifecycle Workflow](./media/image1-7.png)
+
+Developing with containers offers numerous advantages, especially when containers are used in production. The same container configuration is used to host the app in each environment in which it runs, from local dev machine to build and test systems to production. This greatly reduces the likelihood of defects resulting from differences in machine configuration or software versions. Developers can also use whatever tools they're most productive with, including operating system, since containers can run on any OS. In some cases, distributed applications involving many containers may be very resource-intensive to run on a single dev machine. In this scenario, it may make sense to upgrade to using Kubernetes and Azure Dev Spaces, covered in the next section.
+
+As portions of larger applications are broken up into their own smaller, independent *microservices*, additional design patterns can be used to improve app behavior. Instead of working directly with individual services, an *API gateway* can simplify access and decouple the client from its backend. Having separate service backends for different frontends also allows services to evolve in concert with their consumers. Common services can be accessed via a separate *sidecar* container, which might include common client connectivity libraries using the *ambassador* pattern.
+
+![Microservices sample architecture with several common design patterns noted.](./media/image1-10.png)
+
+[Learn more about design patterns to consider when building microservice-based systems.](https://docs.microsoft.com/azure/architecture/microservices/design/patterns)
 
 With Azure handling the management of the nodes in your AKS cluster, you no longer need to perform many tasks manually, like cluster upgrades. Because Azure handles these critical maintenance tasks for you, AKS doesn't provide direct access (such as with SSH) to the cluster.
 
 #### Web App for Containers
```

Comment:
AKS section gone? I guess this is due to the merge from the master that had your previous changes. The title isn't correct - Web App for containers is already covered above. What we need here is the AKS and the section of Dev Spaces below.

Reply:
Ok that was a rather overzealous auto-merge. Fixed.